### PR TITLE
CP-45979 update host-display to python3

### PR DESCRIPTION
**Pull request by acefei**
- format with black
- use xcp.cmd to replace subprocess.popen
Test passed on Xenrt, job id: 3880959

### CODE REVIEW

File: scripts/host-display [hunk 1]
```diff
@@ -91,25 +73,24 @@ def _get_com_port():
     """
     Determine from the current bootloader config which COM port is in use
     """
-    (_, stdout, _) = doexec([XEN_CMDLINE, "--get-xen", "console"])
-    m = re.match(r"console=.*(com\d).*", stdout.readline().strip())
+    (_, stdout) = runCmd([XEN_CMDLINE, "--get-xen", "console"], with_stdout=True)
+    m = re.match(r"console=.*(com\d).*", stdout)
```
Comment (psafont) [inline scripts/host-display:77]: Why isn't the strip needed here anymore?

Comment (acefei) [reply scripts/host-display:77]: I believe the strip is unnecessary as the regex won't capture the trailing whitespaces.

Comment (psafont) [reply scripts/host-display:77]: Strip also removes leading whitespace, which the regex doesn't take into account. (I don't know if it's relevant)

File: scripts/host-display [hunk 2]
```diff
@@ -8,108 +8,96 @@ n.b. the value returned by the "status" command will be correct as of the next
...
+
 def usage():
-    """ Display a usage string. """
-    command_names = list(COMMANDS.keys())
-    command_names.sort()
+    """Display a usage string."""
+    command_names = sorted(COMMANDS.keys())
     print("Usage:")
-    print("    %s {%s}" % (sys.argv[0], '|'.join(command_names)))
+    print("    {} {{{}}}".format(sys.argv[0], "|".join(command_names)))
```
Comment (bernhardkaindl) [inline scripts/host-display:88]: The existing version is shorter and more readable to me:
```suggestion
    print("    %s {%s}" % (sys.argv[0], '|'.join(command_names)))
```
Also, please don't mix actually needed cosmetic changes (like removing spaces aboce) with changes needed Python3, this makes it needlessly hard to review the changes.

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]

File: scripts/host-display (hunk 1 above)
Comment (acefei) [reply scripts/host-display:77]: I tested with this change which works well, but to align with the previous behavior, I will add `strip()`

File: scripts/host-display (hunk 2 above)
Comment (acefei) [reply scripts/host-display:88]: str.format has more performance, but bad readablity in this case, I will revert this change here.

File: scripts/host-display (hunk 1 above)
Comment (acefei) [reply scripts/host-display]: Hi @psafont , I've updated as your comment, could you review again and approve this PR?

**Comment by acefei** [comment]
updated as comments, job 3888721 is passed.

**Review by psafont — APPROVED** [review]
Please squash the last comment into the previous one

File: scripts/host-display [hunk 3]
```diff
@@ -8,108 +8,96 @@ n.b. the value returned by the "status" command will be correct as of the next
...
-        args,
-        stdin=None,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        close_fds=True,
-        universal_newlines=True)
-    rc = proc.wait()
-    stdout = proc.stdout
-    stderr = proc.stderr
-    return (rc, stdout, stderr)
```
Comment (bernhardkaindl) [inline scripts/host-display:45]: By replacing calls to `doexec()` (which calls send_to_syslog) with `xcp.runCmd()` (which does not log the commands to syslog), you remove the logging of the executed commands to the syslog.

Please revert the changes to call `xcp.runCmd()` directly and incorporate `syslog.syslog("%s[%d] - %s" % (SHORT_COMMAND_NAME, os.getpid(), msg))` from the now unused `send_to_syslog()` into the updated `doexec()`.

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]
See my code review comment: Please fix that the new exec function logs like the old did, using the code from the now unused `send_to_syslog()` function,

File: scripts/host-display (hunk 3 above)
Comment (acefei) [reply scripts/host-display]: updated and tested with job id 3889671

**Review by psafont — APPROVED** [review]

**Comment by psafont** [comment]
The commits need to be signed off, this can be done by running `git rebase master --signoff`

**Comment by acefei** [comment]
split commits and tested job id 3895260

**Comment by liulinC** [comment]
job 3895260 passed, PR looks good.
Merge this pR.

**Pull request merged**